### PR TITLE
Add secret-sync provider image support to image-updater tooling

### PR DESCRIPTION
**Pull request by hbhushan3**
https://issues.redhat.com/browse/AROSLSRE-287

1. Semantic Version Sorting 🔧
Fixed a critical bug where the image-updater was selecting older semantic versions over newer ones due to push date ordering. The tool now uses golang.org/x/mod/semver to properly compare semantic versions, ensuring v1.7.2 is correctly prioritized over v1.6.2 regardless of push dates. Includes comprehensive test coverage for version comparison edge cases.

2. Secret-Sync Provider Support 🔑
Added the Azure Key Vault provider image (provider-azure) to automated image tracking. Updated the configuration schema to include providerImage as a containerImage object, and configured the image-updater to automatically monitor and update the secret-sync-controller provider image alongside other tracked components.

Tested secretsync deployment in Personal env:
```
 kubectl get po -n secret-sync-controller
NAME                                                     READY   STATUS    RESTARTS      AGE
secrets-store-sync-controller-manager-5d6c85664b-j6ztx   2/2     Running   1 (44m ago)   46m

$ kubectl get pod -n secret-sync-controller -o jsonpath='{.items[0].spec.containers[?(@.name=="provider-azure-installer")].image}'
mcr.microsoft.com/oss/v2/azure/secrets-store/provider-azure@sha256:9b7a11630793ab772a6e433fc81af45d26838d7682e8412207306adb05e0dc83
```

Tested image bump tool locally:
```
[16:57:35.976] INFO: DRY RUN: Would update image {
  "filePath": "../../config/config.yaml",
  "from": "sha256:9b7a11630793ab772a6e433fc81af45d26838d7682e8412207306adb05e0dc83",
  "jsonPath": "clouds.dev.defaults.secretSyncController.providerImage.digest",
  "line": 875,
  "name": "secretSyncProvider",
  "tag": "v1.7.2",
  "to": "sha256:c34ca57730d55189e843ebfb8343ef6b865e73b80adc3fd43f6181227c1320a7"
}
Checked 1 images, 2 would be updated (dry-run mode)
```

### CODE REVIEW

File: config/config.msft.clouds-overlay.yaml [hunk 1]
```diff
@@ -26,7 +26,8 @@ clouds:
           secretSyncController:
             image:
               digest: sha256:399febaa4c537111589a8a52dea7d1ec4cecce8c3d1010729e875a2e305a2f07 # v0.0.2 (2025-07-17 15:49)
-            providerImage: mcr.microsoft.com/oss/v2/azure/secrets-store/provider-azure:v1.7.0
+            providerImage:
+              digest: sha256:9b7a11630793ab772a6e433fc81af45d26838d7682e8412207306adb05e0dc83
```
Comment (raelga) [inline config/config.msft.clouds-overlay.yaml]: Why it's missing the comment `# v1.7.2 (2025-11-18)`?

**Review by raelga — DISMISSED** [review]

File: config/config.msft.clouds-overlay.yaml (hunk 1 above)
Comment (hbhushan3) [reply config/config.msft.clouds-overlay.yaml]: Let me run the tooling.

Comment (hbhushan3) [reply config/config.msft.clouds-overlay.yaml]: OKay comment is visible now after running the tooling :)

Also deployed to pers env
```
$ kubectl get pod -n secret-sync-controller -o jsonpath='{.items[0].spec.containers[?(@.name=="provider-azure-installer")].image}'
mcr.microsoft.com/oss/v2/azure/secrets-store/provider-azure@sha256:c34ca57730d55189e843ebfb8343ef6b865e73b80adc3fd43f6181227c1320a7
```

**Review by raelga — APPROVED** [review]
